Add "inductor_pre_grad_graph" logging

yushangdi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #142717

Add new structured logging "inductor_pre_grad_graph"

This is for inductor provenance tracking front-end to load this graph from tlparse.

Differential Revision: [D67054578](https://our.internmc.facebook.com/intern/diff/D67054578/)

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

yushangdi committed:
Add "inductor_pre_grad_graph" logging

Add new structured logging "inductor_pre_grad_graph"

This is for inductor provenance tracking front-end to load this graph from tlparse.

Differential Revision: [D67054578](https://our.internmc.facebook.com/intern/diff/D67054578/)

[ghstack-poisoned]

yushangdi committed:
Update on "Add "inductor_pre_grad_graph" logging"


Add new structured logging "inductor_pre_grad_graph"

This is for inductor provenance tracking front-end to load this graph from tlparse.

Differential Revision: [D67054578](https://our.internmc.facebook.com/intern/diff/D67054578/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

yushangdi committed:
Update on "Add "inductor_pre_grad_graph" logging"


Add new structured logging "inductor_pre_grad_graph"

This is for inductor provenance tracking front-end to load this graph from tlparse.

Differential Revision: [D67054578](https://our.internmc.facebook.com/intern/diff/D67054578/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

desertfire commented on torch/_inductor/compile_fx.py:
```diff
@@ -1633,6 +1634,24 @@ def compile_fx(
+            pre_grad_graphs_log.debug(
+                "%s",
+                lazy_format_graph_code(
+                    "BEFORE GRAD",
```
Let's add "PRE" here.

desertfire commented on torch/_logging/_registrations.py:
```diff
@@ -97,6 +97,10 @@
 )
+register_artifact(
+    "pre_grad_graphs",
+    "Prints the FX graph before inductor grad passes. Useful to understand what's being given to Inductor before grad passes",
```
Also better to spell "pre" out here.

Commit:
Update on "Add "inductor_pre_grad_graph" logging"


Add new structured logging "inductor_pre_grad_graph"

This is for inductor provenance tracking front-end to load this graph from tlparse.

Differential Revision: [D67054578](https://our.internmc.facebook.com/intern/diff/D67054578/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

yushangdi commented:
/easycla

desertfire commented:
https://github.com/pytorch/pytorch/pull/137803#pullrequestreview-2363724748  , following this example, did you update tlparse to understand the new traced info?

yushangdi commented:
> [#137803 (review)](https://github.com/pytorch/pytorch/pull/137803#pullrequestreview-2363724748) , following this example, did you update tlparse to understand the new traced info?

@desertfire  Yeah, I've updated the tlparse

yushangdi commented:
/easycla

yushangdi committed:
Update on "Add "inductor_pre_grad_graph" logging"


Add new structured logging "inductor_pre_grad_graph"

This is for inductor provenance tracking front-end to load this graph from tlparse.

Differential Revision: [D67054578](https://our.internmc.facebook.com/intern/diff/D67054578/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

yushangdi commented:
Close because of CLA issue, re-submitted the same PR in https://github.com/pytorch/pytorch/pull/143126
